dcc-portal-ui/app/scripts/advanced/js/advanced.js

```diff
@@ -65,8 +65,34 @@ angular.module('icgc.advanced.controllers', [
 
       _locationFilterCache = _filterService.getCachedFiltersFactory();
 
-      var _isInAdvancedSearchCtrl = true;
+      _controller.donorSets = _.cloneDeep(SetService.getAllDonorSets());
+      _controller.geneSets = _.cloneDeep(SetService.getAllGeneSets());
+      _controller.mutationSets = _.cloneDeep(SetService.getAllMutationSets());
+
+      // to check if a set was previously selected and if its still in effect
+      const checkSetInFilter = (entity, entitySets) => {
+        let filters = _locationFilterCache.filters();
+        const setIdPrefix = 'ES:';
+
+        if(filters[entity] && filters[entity].id){
+          _.each(entitySets, (set) => {
+            set.selected = false;
+            _.each(filters[entity].id.is, (id) => {
+              if(_.includes(id, setIdPrefix)){
+                if(`${setIdPrefix}${set.id}` === id){
+                  set.selected = true;
+                }
+              }
+            });
+          });
+        } else {
+          _.each(entitySets, (set) => {
+            set.selected = false;
+          });
+        }
+      };
```

Comment:
I would recommend somethign like updateSetsSelection

Also does the code simplification I recommended work?
It's not really about the lines of code but rather complexity - how many lines of code do you have to read and mentally execute to figure out what a function does